File: 1395/CountNumberOfTeams.cpp
/*
https://leetcode.com/problems/count-number-of-teams/description/?envType=daily-question&envId=2024-07-29
*/

class Solution {
public:
    int numTeams(vector<int>& rating) {
        int sz=rating.size(), res=0;
        for(int i=0; i<sz; ++i){
            int curr=rating[i];
            if(i+1<=sz-1){
                for(int j=i+1; j<sz; ++j){
                    int next=rating[j];
                    if(next>curr){
                        if(j+1<=sz-1){
                            for(int k=j+1; k<sz; ++k){
                                if(rating[k]>next){
                                    res++;
                                }
                            }
                        }
                    } else if(next<curr){
                        if(j+1<=sz-1){
                            for(int k=j+1; k<sz; ++k){
                                if(rating[k]<next){
                                    res++;
                                }
                            }
                        }
                    }
                }
            }
        }
        return res;
    }
};

File: 1/1395/CountNumberOfTeams.py
# https://leetcode.com/problems/count-number-of-teams/description/?envType=daily-question&envId=2024-07-29

class Solution:
    def numTeams(self, rating: List[int]) -> int:
        lengthOfRating=len(rating)
        resultantNumberOfTeams=0
        for i in range(0, lengthOfRating, 1):
            curr=rating[i]
            if i+1<=lengthOfRating-1:
                for j in range(i+1, lengthOfRating, 1):
                    next=rating[j]
                    if next>curr:
                        if j+1<=lengthOfRating-1:
                            for k in range(j+1, lengthOfRating, 1):
                                if rating[k]>next:
                                    resultantNumberOfTeams+=1
                    elif next<curr:
                        if j+1<=lengthOfRating-1:
                            for k in range(j+1, lengthOfRating, 1):
                                if rating[k]<next:
                                    resultantNumberOfTeams+=1
        return resultantNumberOfTeams